--- origsrc/libwmf-0.2.8.4/libwmf-config.in	2004-02-10 06:41:26.000000000 -0600
+++ src/libwmf-0.2.8.4/libwmf-config.in	2006-10-19 22:54:55.944500000 -0500
@@ -2,19 +2,8 @@
 
 exec_prefix_set=no
 
-case `uname` in
-  CYGWIN*)
-    prefix=`bash -c "type -p libwmf-config"`
-    prefix=`dirname $prefix`
-    prefix=`dirname $prefix`
-    prefix=`cygpath -m "$prefix"`
-    exec_prefix="$prefix"
-  ;;
-  *)
-    prefix=@prefix@
-    exec_prefix=@exec_prefix@
-  ;;
-esac
+prefix=@prefix@
+exec_prefix=@exec_prefix@
 
 usage()
 {
@@ -96,12 +85,6 @@
 wmf_cflags="@WMF_CONFIG_CFLAGS@"
 wmf_liblflags="@WMF_LIBFLAGS@"
 
-case `uname` in
-  CYGWIN*)
-    wmf_cflags=`echo $wmf_cflags | sed -e "s%@prefix@%$prefix%"`
-    wmf_liblflags=`echo $wmf_liblflags | sed -e "s%@prefix@%$prefix%"`
-  ;;
-esac
 
 libwmf_buildstyle=@LIBWMF_BUILDSTYLE@
 if test $libwmf_buildstyle = heavy; then
--- origsrc/libwmf-0.2.8.4/src/Makefile.am	2004-02-10 06:41:26.000000000 -0600
+++ src/libwmf-0.2.8.4/src/Makefile.am	2006-10-19 22:46:04.382750000 -0500
@@ -20,7 +20,7 @@
 
 SUBDIRS = $(DIRHEAVY)
 
-CFLAGS = @CFLAGS@ @WMF_CFLAGS@ @GDK_PIXBUF_CFLAGS@
+AM_CFLAGS = @WMF_CFLAGS@ @GDK_PIXBUF_CFLAGS@
 
 DEFS = @DEFS@ @WMF_DEFS@ $(DEFHEAVY) $(GDK_PIXBUF_DEFS) \
 	-DPACKAGE=\"@PACKAGE@\" \
--- origsrc/libwmf-0.2.8.4/src/convert/Makefile.am	2002-05-16 02:48:38.000000000 -0500
+++ src/libwmf-0.2.8.4/src/convert/Makefile.am	2006-10-19 22:46:04.726500000 -0500
@@ -1,6 +1,6 @@
-CFLAGS = @CFLAGS@ @WMF_CFLAGS@
+AM_CFLAGS = @WMF_CFLAGS@
 
-LDFLAGS = -L$(top_builddir)/src/.libs @WMF_LIBFLAGS@ @LDFLAGS@
+LDADD = -L$(top_builddir)/src/.libs @WMF_LIBFLAGS@ @LDFLAGS@
 
 DEFS = @DEFS@ @WMF_DEFS@ \
 	-DPACKAGE=\"@PACKAGE@\" \
--- origsrc/libwmf-0.2.8.4/src/extra/gd/Makefile.am	2002-06-23 03:48:23.000000000 -0500
+++ src/libwmf-0.2.8.4/src/extra/gd/Makefile.am	2006-10-19 22:46:05.054625000 -0500
@@ -1,8 +1,6 @@
 INCLUDES = -I$(srcdir)
 
-CFLAGS = @CFLAGS@ @WMF_CFLAGS@
-
-LDFLAGS = @LDFLAGS@
+AM_CFLAGS = @WMF_CFLAGS@
 
 DEFS = @DEFS@ @GD_DEFS@
 
--- origsrc/libwmf-0.2.8.4/src/extra/trio/Makefile.am	2001-07-11 05:12:27.000000000 -0500
+++ src/libwmf-0.2.8.4/src/extra/trio/Makefile.am	2006-10-19 22:46:05.757750000 -0500
@@ -1,6 +1,4 @@
-CFLAGS = @CFLAGS@ @WMF_CFLAGS@
-
-LDFLAGS = @LDFLAGS@
+AM_CFLAGS = @WMF_CFLAGS@
 
 DEFS = @DEFS@ @WMF_DEFS@
 
--- origsrc/libwmf-0.2.8.4/src/ipa/Makefile.am	2002-05-16 02:48:10.000000000 -0500
+++ src/libwmf-0.2.8.4/src/ipa/Makefile.am	2006-10-19 22:46:05.773375000 -0500
@@ -1,6 +1,4 @@
-CFLAGS = @CFLAGS@ @WMF_CFLAGS@
-
-LDFLAGS = @LDFLAGS@
+AM_CFLAGS = @WMF_CFLAGS@
 
 DEFS = @DEFS@ @WMF_DEFS@ \
 	-DPACKAGE=\"@PACKAGE@\" \
--- origsrc/libwmf-0.2.8.4/src/player.c	2002-12-10 13:30:26.000000000 -0600
+++ src/libwmf-0.2.8.4/src/player.c	2006-10-19 22:46:05.789000000 -0500
@@ -42,6 +42,7 @@
 #include "player/defaults.h" /* Provides: default settings               */
 #include "player/record.h"   /* Provides: parameter mechanism            */
 #include "player/meta.h"     /* Provides: record interpreters            */
+#include <stdint.h>
 
 /**
  * @internal
@@ -132,8 +133,14 @@
 		}
 	}
 
-/*	P->Parameters = (unsigned char*) wmf_malloc (API,(MAX_REC_SIZE(API)-3) * 2 * sizeof (unsigned char));
- */	P->Parameters = (unsigned char*) wmf_malloc (API,(MAX_REC_SIZE(API)  ) * 2 * sizeof (unsigned char));
+	if (MAX_REC_SIZE(API) > UINT32_MAX / 2)
+	{
+		API->err = wmf_E_InsMem;
+		WMF_DEBUG (API,"bailing...");
+		return (API->err);
+	}
+	
+ 	P->Parameters = (unsigned char*) wmf_malloc (API,(MAX_REC_SIZE(API)  ) * 2 * sizeof (unsigned char));
 
 	if (ERR (API))
 	{	WMF_DEBUG (API,"bailing...");
--- origsrc/libwmf-0.2.8.4/wmfconfig.h.in	2005-07-27 16:46:54.000000000 -0500
+++ src/libwmf-0.2.8.4/wmfconfig.h.in	2006-10-19 23:16:02.391750000 -0500
@@ -132,10 +132,10 @@
 /* Define to empty if `const' does not conform to ANSI C. */
 #undef const
 
-/* Define to `long' if <sys/types.h> does not define. */
+/* Define to `long int' if <sys/types.h> does not define. */
 #undef off_t
 
-/* Define to `unsigned' if <sys/types.h> does not define. */
+/* Define to `unsigned int' if <sys/types.h> does not define. */
 #undef size_t
 
 /* define to your implementation of snprintf */
